klippy/extras/delta_calibrate.py

```diff
@@ -6,11 +6,107 @@
 import math, logging
 import probe, mathutil
 
+
+######################################################################
+# Delta "stable position" coordinates
+######################################################################
+
+# A "stable position" is a 3-tuple containing the number of steps
+# taken since hitting the endstop on each delta tower.  Delta
+# calibration uses this coordinate system because it allows a position
+# to be described independent of the software parameters.
+
+# Return cartesian coordinates for the given stable_positions when the
+# given delta_params are used.
+def get_position_from_stable(stable_position, delta_params):
+    dp = delta_params
+    sphere_coords = [
+        (t[0], t[1], es - sp * sd)
+        for sd, t, es, sp in zip(
+                dp.stepdists, dp.towers, dp.abs_endstops, stable_position) ]
+    return mathutil.trilateration(sphere_coords, [a**2 for a in dp.arms])
+
+# Return a stable position from the nominal delta tower positions
+def get_stable_position(stepper_position, delta_params):
+    dp = delta_params
+    return [int((ep - sp) / sd + .5)
+            for sd, ep, sp in zip(
+                    dp.stepdists, dp.abs_endstops, stepper_position)]
+
+# Return a stable position from a cartesian coordinate
+def calc_stable_position(coord, delta_params):
+    dp = delta_params
+    steppos = [
+        math.sqrt(a**2 - (t[0]-coord[0])**2 - (t[1]-coord[1])**2) + coord[2]
+        for t, a in zip(dp.towers, dp.arms) ]
+    return [(ep - sp) / sd
+            for sd, ep, sp in zip(dp.stepdists, dp.abs_endstops, steppos)]
+
+
+######################################################################
+# Delta calibration object
+######################################################################
+
+# The angles and distances of the calibration object found in
+# docs/prints/calibrate_size.stl
+MeasureAngles = [210., 270., 330., 30., 90., 150.]
+MeasureOuterRadius = 65
+MeasureRidgeRadius = 5. - .5
+
+# How much to prefer a distance measurement over a height measurement
+MEASURE_WEIGHT = 0.5
+
+# Convert distance measurements made on the calibration object to
+# 3-tuples of (actual_distance, stable_position1, stable_position2)
+def measurements_to_distances(measured_params, delta_params):
+    # Extract params
+    mp = measured_params
+    dp = delta_params
+    scale = mp['SCALE']
```

Comment:
Here scale becomes a list and fails later be multiplied by MeasureRidgeRadius on line 77, resulting in exception